src/ctia/main.clj

```diff
@@ -1,10 +1,9 @@
 (ns ctia.main
-  (:gen-class)
-  (:require
-   [ctia.init :refer [log-properties start-ctia!]]))
+  (:gen-class))
 
 (defn -main
   "Application entry point"
   [& args]
-  (start-ctia! :join? true
-               :silent? false))
+  (let [app ((requiring-resolve 'ctia.init/start-ctia!))]
```

Comment:
I don't get the subtlety of using `requiring-resolve` here instead of using using a classic `require` here, for both `start-ctia!` and `run-app`.

Reply:
It's so then the `gen-class` directive generates a minimum number of classes, this is like trapperkeeper's main: https://github.com/puppetlabs/trapperkeeper/blob/master/src/puppetlabs/trapperkeeper/main.clj

Reply:
Going an additional layer of why, this makes REPL development slightly easier. There was a discussion about it a few months ago in CTIA https://github.com/threatgrid/ctia/pull/918#issuecomment-659604548